src/linear_algebra/affine_space/affine_equiv.lean

```diff
@@ -160,6 +160,49 @@ protected lemma bijective (e : P₁ ≃ᵃ[k] P₂) : bijective e := e.to_equiv.
 protected lemma surjective (e : P₁ ≃ᵃ[k] P₂) : surjective e := e.to_equiv.surjective
 protected lemma injective (e : P₁ ≃ᵃ[k] P₂) : injective e := e.to_equiv.injective
 
+/-- Bijective affine maps are affine isomorphisms. -/
+noncomputable def of_bijective {R V₁ V₂ P₁ P₂ : Type*}
+  [ring R] [add_comm_group V₁] [add_comm_group V₂] [module R V₁]
+  [module R V₂]
+  [add_torsor V₁ P₁] [add_torsor V₂ P₂]
+  {φ : P₁ →ᵃ[R] P₂}
+  (hφ : function.bijective φ) : P₁ ≃ᵃ[R] P₂ :=
+begin
+  refine ⟨equiv.of_bijective _ hφ, _, _⟩,
+  { refine linear_equiv.of_bijective φ.linear _ _ ;
+      obtain ⟨_, _⟩ := hφ ;
+      simp only [φ.injective_iff_linear_injective, φ.surjective_iff_linear_surjective] ;
```

Comment:
I think there's a `φ.bijective_iff_linear_bijective` lemma hiding here? you should be able to prove it in one line with `and_congr`.

Reply:
I guess I could write `bijective_iff_linear_bijective` if it is useful elsewhere, but how would it be used in this lemma? `linear_equiv.of_bijective` doesn't take a `bijective` argument but two `injective` and `surjective` arguments, so I'd have to decompose `φ.bijective_iff_linear_bijective.mpr hφ` into those components again...

```lean
lemma of_bijective_linear {R V₁ V₂ P₁ P₂ : Type*}
  [ring R] [add_comm_group V₁] [add_comm_group V₂] [module R V₁]
  [module R V₂]
  [add_torsor V₁ P₁] [add_torsor V₂ P₂]
  {φ : P₁ →ᵃ[R] P₂}
  (hφ : function.bijective φ) :
  (of_bijective hφ).linear = linear_equiv.of_bijective φ.linear
    (φ.bijective_iff_linear_bijective.mpr hφ).1
    (φ.bijective_iff_linear_bijective.mpr hφ).2 := rfl
```

Reply:
I've opened #17897 to tidy up `linear_equiv.of_bijective`, well spotted.